File: Python/2645. Minimum Additions to Make Valid String.py
class Solution:
    def addMinimum(self, word: str) -> int:
        abc, pa, ps = "abc", 0, 0
        res = 0

        while ps < len(word):
            if abc[pa] == word[ps]: pa = (pa + 1) % 3; ps += 1
            else: res += 1; pa = (pa + 1) % 3

        return res + (3 - pa if pa != 0 else 0)

File: C/2645. Minimum Additions to Make Valid String.c
int addMinimum(char * word){
    char* abc = "abc";
    int pa = 0, ps = 0, res = 0, l = strlen(word);

    while (ps < l) {
        if (abc[pa] == word[ps]) ps++;
        else res++;

        pa = (pa + 1) % 3;
    }

    return res + (pa == 0 ? 0 : 3 - pa);
}
